docs/development/hacking_itk_wasm.md

```diff
@@ -1,69 +1,252 @@
+
+We follow [standard GitHub contribution best practices]: pull requests made from forks are tested with continuous integration tests, and updates are made via code review before integration.
+
+We use the [conventional commit] standard for our commit message format in the *itk-wasm* repository.
 
-- [Node.js / NPM](https://nodejs.org/en/download/)
-- [Docker](https://docs.docker.com/install/)
-  * On Linux, make sure you can run [`docker` without `sudo`](https://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo).
-  * On Windows, we recommend [WSL 2 with Docker enabled](https://docs.docker.com/desktop/windows/wsl/).
-- Bash
+The C++ core and Docker build environment can be developed independently from the CLI, language-specific libraries, and example packages. The latter are developed as a [pnpm workspace]. Individual packages in the workspace can be developed independently after `pnpm install` is executed at the root.
 
-## Building
+## C++ core
+
+ITK-Wasm's [C++ core](../introduction/parts.md#cxx-core) can be developed with native toolchains. The steps are:
+
+0. Install a [C++ compiler toolchain] and [CMake]
+1. Build [ITK]
+2. Build the `WebAssemblyInterface` module from this module against ITK
+3. Run the tests
+
+From a command line, these steps in brief are:
 
-To build itk-wasm itself:
 ```sh
-npm install
-npm run build
+# Build ITK
+git clone https://github.com/InsightSoftwareConsortium/ITK
+cmake -BITK-build -SITK -DBUILD_TESTING=OFF -DModule_MeshToPolyData=ON
+cmake --build ITK-build
+
+# Build the WebAssemblyInterface module from this module against ITK
+git clone https://github.com/InsightSoftwareConsortium/itk-wasm
+cmake -Bitk-wasm-build -Sitk-wasm -DBUILD_TESTING=ON -DITK_DIR=$PWD/ITK-build
+cmake --build itk-wasm-build
+
+# Run the tests
+ctest --test-dir itk-wasm-build
 ```
 
-Run the tests:
+For additional guidance on C++ development, see the [ITK Software Guide].
+
+### Testing data
+
+To prevent Git repository bloat, we add testing data through [CMake content links](https://docs.itk.org/en/latest/contributing/upload_binary_data.html) of [Content Identifiers (CIDs)](https://proto.school/anatomy-of-a-cid). To add new test data,
+
+1. Upload the data and download its content link *.cid* file with the [CMake w3 ExternalData Upload Tool](https://content-link-upload.itk.org/)
+2. Move the *.cid* file to *itk-wasm/test/Input/*
+3. Reference the content link with a `DATA{<path>}` call in *itk-wasm/test/CMakeLists.txt*.
+
+## Build environment Docker images
+
+Builds of the [wasm build environment Docker images](../introduction/parts.md#docker) or WebAssembly modules require [Docker](https://docs.docker.com/install/).
+
+* On Linux, make sure you can run [`docker` without `sudo`](https://askubuntu.com/questions/477551/how-can-i-use-docker-without-sudo).
+* On Windows, we recommend [WSL 2 with Docker enabled](https://docs.docker.com/desktop/windows/wsl/).
```

Comment:
👍 +1 for WSL2. We can install and use Docker Engine (the free kind), without difficulty. I haven't faced any significant issues developing with itk-wasm on WSL2.

Reply:
Good to know!

I will add a note about Docker Desktop licensing, too.